### ISSUE TITLE
pass:text fails for text colour that is a temp vec

### ISSUE THREAD

**Issue by eigenbom** [bjornbytes/lovr#614]
The following code fails:

```lua
local c = lovr.math.vec4(1,1,1,1) 
pass:text({c,"text"}, ...)
```

with error `Error: Expected nil, number, table, vec3, or vec4 for color value`

I think it's because luax_optcolor isn't temp vec aware: https://github.com/bjornbytes/lovr/blob/efc81e4cec9629483d3541a34d17eeb05592e040/src/api/api.c#L441

**Comment by bjornbytes** [comment]
Fixed in a79b3af7e5fdddf43cb5e86a557b7811eb22e7b9.  Thanks!

**Comment by eigenbom** [comment]
Did you fix this within 4 minutes of me reporting it?! ⚡

Thanks :)
